Sources/AlgoliaSearchClient/Transport/RetryStrategy/AlgoliaRetryStrategy.swift

```diff
@@ -37,7 +37,10 @@ class AlgoliaRetryStrategy: RetryStrategy {
       }
 
       return HostIterator { [weak self] in
-        self?.hosts.first { $0.supports(callType) && $0.isUp }
+        guard let retryStrategy = self else { return nil }
```

Comment:
Question: If I understood correctly `guard` is the `mutex/lock` of Swift, if it works like inc#/c++ shouldn't it be at the instance level to protect the instance of multiple threads?

Reply:
`guard` is the statement, ensuring the fulfilment of a condition in the scope bellow. 
```swift
let condition: Bool

if condition {
 // here the condition is true 
} else {
 // here the condition is false
}

guard condition else {
  // here the condition is false
}
// here the condition is true
``` 

In other words, it "protects" the code bellow from non-fulfilment of a logical expression. 
It works the same way with optional variables, unwrapping them for the code bellow. 
Here the `guard` statement ensures the code bellow that weakly captured optional `self` (equivalent of `this` in Java) is not `nil`. 

`retryStrategy.queue` is the queue used as a `lock` for access to `retryStrategy.hosts` list. It is actually present at the instance level of `AlgoliaRetryStrategy`.

Reply:
Thanks for the explanation 👍🏻